node_shell.py

```diff
@@ -387,10 +851,271 @@ def do_exit(self, arg):  # pylint: disable=unused-argument
+            # No action needed
+            pass
+        except Exception as exception:
+            # Catch all python exceptions raised during the verdi command execution
+            self.poutput(
+                'Verdi Command raised an exception {}'.format(exception))
+            print_exc(file=self.stdout)
+
+    group_parser = cmd2.Cmd2ArgumentParser()
+
+    group_parser.add_argument('--user-email', '-u', help='Filter by user')
+    group_parser.add_argument('--all-users',
+                              help='Flag if include all users',
+                              action='store_true')
+    group_parser.add_argument('--all-types',
+                              '-a',
+                              help='Flag if include all types',
+                              action='store_true')
+    group_parser.add_argument('--group-type',
+                              '-t',
+                              default=GroupTypeString.USER.value,
+                              help='Filter by type')
+    group_parser.add_argument('--with-description',
+                              '-d',
+                              help='Show also the decription',
+                              action='store_true')
+    group_parser.add_argument('--startswith',
+                              '-s',
+                              help='Filter by the initial string')
+    group_parser.add_argument('--endswith',
+                              '-e',
+                              help='Filter by ending string')
+    group_parser.add_argument(
+        '--contains',
+        '-c',
+        help='Filter by checking if the group name contains the string')
+    group_parser.add_argument(
+        '--past-days',
+        '-p',
+        type=int,
+        help='Filter by only including those created in the past N days')
+    group_parser.add_argument(
+        '--count',
+        '-C',
+        action='store_true',
+        help='Show also the number of nodes in the group')
+
+    @cmd2.with_argparser(group_parser)
+    def do_group_list(self, args):
+        """Command for list groups"""
+        from aiida.cmdline.commands.cmd_group import group_list
+        with self.verdi_isolate():
+            group_list.callback(all_users=args.all_users,
+                                user_email=args.user_email,
+                                all_types=args.all_types,
+                                group_type=args.group_type,
+                                with_description=args.with_description,
+                                count=args.count,
+                                past_days=args.past_days,
+                                startswith=args.startswith,
+                                endswith=args.endswith,
+                                contains=args.contains,
+                                node=None)
+
+    @cmd2.with_argparser(group_parser)
+    @needs_node
+    def do_group_belong(self, args):
```

Comment:
Here maybe I would just call the internal AiiDA command (or directly run a query) without providing all these options? I.e. just provide the list of groups this node is in, simplifying the number of options that you can use?